tests/unit-core/test-arraybuffer.c

```diff
@@ -154,15 +154,26 @@ static void test_write_with_offset (uint8_t offset) /**< offset for buffer write
 static bool callback_called = false;
 static bool detach_free_callback_called = false;
 
-static void test_free_cb (void *buffer) /**< buffer to free (if needed) */
+static void
+test_free_cb (uint32_t length, /**< length of buffer */
+              void *buffer, /**< buffer to free (if needed) */
+              void *user_p) /**< user pointer */
 {
+  JERRY_UNUSED (length);
+  JERRY_UNUSED (user_p);
+
   (void) buffer;
   callback_called = true;
 } /* test_free_cb */
 
-static void test_detach_free_cb (void *buffer) /**< buffer to free */
+static void test_detach_free_cb (uint32_t length, /**< length of buffer */
+                                 void *buffer, /**< buffer to free (if needed) */
+                                 void *user_p) /**< user pointer */
 {
-  free (buffer);
+  JERRY_UNUSED (length);
+  JERRY_UNUSED (user_p);
+  JERRY_UNUSED (buffer);
```

Comment:
This seems strange: if the "free" call is removed where will be the "buffer" freed?